server.go

```diff
@@ -199,6 +199,9 @@ type server struct {
 	peerConnectedListeners    map[string][]chan<- lnpeer.Peer
 	peerDisconnectedListeners map[string][]chan<- struct{}
 
+	// TODO(yy): the bool value seems to be unused, we ask the connmgr to
+	// make a permanent connection regardless of this value. Needs further
+	// check.
```

Comment:
It is used in `prunePersistentPeerConnection` to ensure that we dont prune a peer we have marked as permanent

Reply:
the bool represents if we should/shouldnt keep trying to reconnect with the peer even if we have no active channels to them. But i think you might be right in that it isnt actually used properly... 
[Here](https://github.com/lightningnetwork/lnd/blob/master/peer/brontide.go#L495) we prune the connection if num active channels is 0
which then will call [this](https://github.com/lightningnetwork/lnd/blob/master/server.go/#L2851) which remove this peer from the `peristenetPeers` map _and_ cancels connection reqs to it. but then back in the Brontide Start method, it just continues as per usual even though we may now have canceled the connReq to the peer.... hmmm... do you think this is a bug?

Reply:
Yeah I think it's a bug without a trivial fix...Worst case tho is an error will be returned from that Start method. Thb it's not easy to follow the code. Speaking of which, just curious, any thoughts on how to refactor this peer conn?

Reply:
> any thoughts on how to refactor this peer conn?

I guess just continuing with #5700 & then handling the returned value from `prunePersistentPeerConnection` properly. Or do you mean something else?